pkg/csi/cinder/controllerserver.go

```diff
@@ -944,7 +945,28 @@ func (cs *controllerServer) ValidateVolumeCapabilities(ctx context.Context, req
 }
 
 func (cs *controllerServer) GetCapacity(ctx context.Context, req *csi.GetCapacityRequest) (*csi.GetCapacityResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "GetCapacity is not yet implemented")
+	klog.V(4).Infof("GetCapacity: called with args %+v", protosanitizer.StripSecrets(*req))
+
+	topology := req.GetAccessibleTopology()
+	if topology != nil {
+		region := topology.GetSegments()[corev1.LabelTopologyRegion]
+
+		cloud, cloudExist := cs.Clouds[region]
+		if !cloudExist {
+			return nil, status.Error(codes.InvalidArgument, "[GetCapacity] specified cloud undefined")
+		}
+
+		availableCapacity, err := cloud.GetFreeQuotaStorageSpace()
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "[GetCapacity]: failed with error %v", err)
+		}
+
+		return &csi.GetCapacityResponse{
+			AvailableCapacity: int64(availableCapacity * 1024 * 1024 * 1024),
+		}, nil
+	}
+
+	return &csi.GetCapacityResponse{}, nil
```

Comment:
https://refactoring.guru/es/replace-nested-conditional-with-guard-clauses

```suggestion
	topology := req.GetAccessibleTopology()
	if topology == nil {
		return &csi.GetCapacityResponse{}, nil
	}
	
	region := topology.GetSegments()[corev1.LabelTopologyRegion]
	cloud, cloudExist := cs.Clouds[region]
	if !cloudExist {
		return nil, status.Error(codes.InvalidArgument, "[GetCapacity] specified cloud undefined")
	}

	availableCapacity, err := cloud.GetFreeQuotaStorageSpace()
	if err != nil {
		return nil, status.Errorf(codes.Internal, "[GetCapacity]: failed with error %v", err)
	}

	return &csi.GetCapacityResponse{
		AvailableCapacity: int64(availableCapacity * 1024 * 1024 * 1024),
	}, nil
```